src/scalardl/src/main/java/com/example/contract/token/ExchangeToken.java

```diff
@@ -0,0 +1,62 @@
+package com.example.contract.token;
+
+import com.scalar.dl.ledger.asset.Asset;
+import com.scalar.dl.ledger.contract.Contract;
+import com.scalar.dl.ledger.database.Ledger;
+import com.scalar.dl.ledger.exception.ContractContextException;
+import java.util.Optional;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+public class ExchangeToken extends Contract {
+
+  @Override
+  public JsonObject invoke(Ledger ledger, JsonObject argument, Optional<JsonObject> properties) {
+    if (!argument.containsKey(Const.KEY_SRC_USER_ID)
+        || !argument.containsKey(Const.KEY_DST_USER_ID)
+        || !argument.containsKey(Const.KEY_AMOUNT)
+        || !argument.containsKey(Const.KEY_EXCHANGE_RATE)) {
+      throw new ContractContextException(
+          "Please set "
+              + Const.KEY_SRC_USER_ID
+              + ", "
+              + Const.KEY_DST_USER_ID
+              + ", "
+              + Const.KEY_AMOUNT
+              + " and "
+              + Const.KEY_EXCHANGE_RATE
+              + " in the argument");
+    }
+
+    String srcUserId = Integer.toString(argument.getInt(Const.KEY_SRC_USER_ID));
+    String dstUserId = Integer.toString(argument.getInt(Const.KEY_DST_USER_ID));
+    int amount = argument.getInt(Const.KEY_AMOUNT);
+    double exchangeRate = argument.getJsonNumber(Const.KEY_EXCHANGE_RATE).doubleValue();
```

Comment:
Using `exchangeRate` is the only difference in between `ExchangeToken` and `SendToken` ?

Reply:
Yes. ExchangeToken exchanges tokens among users in different domain. But currently it does not check anything related to the domain to keep it simple. If there are any requirements for our desired workload to reproduce, I will add.